pkg/controller/statusmanager/status_manager.go

```diff
@@ -63,6 +67,39 @@ func (status *StatusManager) set(reachedAvailableLevel bool, conditions ...confi
 			return err
 		}
 
+		// Check for differences between current and rendered related objects, delete if necessary
+		var found bool = false
+		for _, current_obj := range co.Status.RelatedObjects {
+			for _, rendered_obj := range status.relatedObjects {
+				found = reflect.DeepEqual(current_obj, rendered_obj)
+
+				if found {
+					break
+				}
+			}
+			if !found {
+				log.Printf("Detected related object not rendered by manifests, deleting...")
+				gvr := schema.GroupVersionResource{
+					Group:    current_obj.Group,
+					Resource: current_obj.Resource,
+				}
+				gvk, err := status.mapper.KindFor(gvr)
+				if err != nil {
+					log.Printf("Error getting GVK of object for deletion: %v", err)
+					continue
+				}
+				obj_to_delete := &uns.Unstructured{}
+				obj_to_delete.SetName(current_obj.Name)
+				obj_to_delete.SetNamespace(current_obj.Namespace)
+				obj_to_delete.SetGroupVersionKind(gvk)
+				status.client.Delete(context.Background(), obj_to_delete)
```

Comment:
2 things.
- you still need to `err = Delete()`
- This isn't going to trigger a background delete. We might not have the ability to request a background delete via our client, so let's just ignore that.

Reply:
Also, a quick check shows that we'll have the ability to issue a Background delete when 336 merges.

Reply:
336 merged, and added missing err var, marking as resolved